pandas/io/pytables.py

```diff
@@ -806,9 +803,10 @@ def select_as_coordinates(
         stop  : integer (defaults to None), row number to stop selection
         """
         where = _ensure_term(where, scope_level=1)
-        return self.get_storer(key).read_coordinates(
-            where=where, start=start, stop=stop, **kwargs
-        )
+        tbl = self.get_storer(key)
+        if not isinstance(tbl, Table):
```

Comment:
Does doing this solve a bug?

Reply:
mypy mostly.  get_storer can return two different types, and this method only works if we get a Table back